web/impact/impact/v1/views/mentor_program_office_hour_list_view.py

```diff
@@ -0,0 +1,60 @@
+# MIT License
+# Copyright (c) 2019 MassChallenge, Inc.
+from django.db.models import Value as V
+from django.db.models.functions import Concat
+
+from impact.v1.views.base_list_view import BaseListView
+from impact.v1.helpers import (
+    MentorProgramOfficeHourHelper,
+)
+
+ID_FIELDS = ['mentor_id', 'finalist_id']
+NAME_FIELDS = ['mentor_name', 'finalist_name']
+
+
+class MentorProgramOfficeHourListView(BaseListView):
+    view_name = "office_hour"
+    helper_class = MentorProgramOfficeHourHelper
+
+    def filter(self, qs):
+        if not self.request.query_params.keys():
+            return qs
+
+        if self._has_participant_filter(NAME_FIELDS):
+            return self._filter_by_participant_name(qs)
+
+        if self._has_participant_filter(ID_FIELDS):
+            param_items = self.request.query_params.dict().items()
+            return self._filter_by_participant_id(qs, param_items)
+
+    def _filter_by_participant_name(self, qs):
+        params = self.request.query_params
+        mentor_name = params.get('mentor_name', None)
+        finalist_name = params.get('finalist_name', None)
+
+        if mentor_name:
+            return self._generate_name_query(qs, 'mentor', mentor_name)
+        if finalist_name:
+            return self._generate_name_query(qs, 'finalist', finalist_name)
+        return qs.none()
```

Comment:
this function implies we can't filter by both `mentor_name` **and** `finalist_name` since the function terminates if there's a mentor name. is it expected that we only filter by one of them?

Reply:
Yes that is the case. It wasn't spec'ed out in the ticket. I do not envision a use case for filtering by both `mentor_name` and `finalist_name`. Remember the use case here would be searching. In cases where we'd like to retrieve office hours for  a specific mentor and finalist, their ids would be the most appropriate. cc @shanbady, @hfreund

Reply:
Herman's position seems reasonable to me.

Reply:
#drive-by 
It seems like it would not be hard to make this a bit more flexible now, which would save us a lot of complaining later on when we decide we actually do want to have that flexibility. 
(when I say "not hard", I think the function actually gets simpler, since it can be a loop or comprehension)

Reply:
thanks for the change..one more thing here. I believe the condition now as is is `if the mentor has name A and the menteee has name B then get that entry`, I think the condition should instead be `or` to make this a bit more flexible

Reply:
done